diff --git a/package.json b/package.json
index 5152843..e38029d 100644
--- a/package.json
+++ b/package.json
@@ -2,7 +2,7 @@
   "name": "dilithium-crystals-js",
   "version": "1.1.3",
   "main": "src/node/index.js",
-  "browser": "dist/dilithium.min.js",
+  "browser": "src/browser/index.js",
   "files": [
     "src/**/*.js",
     "src/**/*.d.ts",
diff --git a/src/browser/index.js b/src/browser/index.js
index 9e63e2a..7586ca4 100644
--- a/src/browser/index.js
+++ b/src/browser/index.js
@@ -1,5 +1,6 @@
 import dilithiumFactory from "../../dilithium.js";
-import { DILITHIUM_PARAMS, generateKeys, sign, verify } from "../common";
+import common from "../common.js";
+const { DILITHIUM_PARAMS, generateKeys, sign, verify } = common;
 
 // Function to fetch the WebAssembly file
 async function fetchWasm() {
@@ -32,4 +33,5 @@ const createDilithium = () =>
       DILITHIUM_PARAMS,
     }));
 
-export { createDilithium };
+export { createDilithium };
+export default createDilithium();
diff --git a/src/node/index.js b/src/node/index.js
index a4f192e..536bb00 100644
--- a/src/node/index.js
+++ b/src/node/index.js
@@ -13,7 +13,7 @@ try {
   wasmBinary = fs.readFileSync(wasmPath);
 } catch (err) {
   // If that fails, try the Electron production path
-  wasmPath = path.join(process.resourcesPath, "assets", "dilithium.wasm");
+  wasmPath = path.join(typeof process.resourcesPath !== "undefined" ? process.resourcesPath : process.cwd(), "assets", "dilithium.wasm");
   wasmBinary = fs.readFileSync(wasmPath);
 }
 
